test/jdk/jdk/jfr/threading/TestDeepVirtualStackTrace.java

```diff
@@ -92,32 +92,50 @@ private static void assertStackTrace(Runnable eventEmitter, String eventName, St
         System.out.println();
         System.out.println("Testing event: " + eventName);
         System.out.println("=============================");
+
+        boolean isTargetEventFound = false;
+
         try (Recording r = new Recording()) {
             r.enable(eventName).withoutThreshold();
             r.start();
             Thread vt = Thread.ofVirtual().start(eventEmitter);
             vt.join();
             r.stop();
             List<RecordedEvent> events = Events.fromRecording(r);
-            Asserts.assertEquals(events.size(), 1, "No event found in virtual thread");
-            RecordedEvent event = events.get(0);
-            System.out.println(event);
-            RecordedStackTrace stackTrace = event.getStackTrace();
-            List<RecordedFrame> frames = stackTrace.getFrames();
-            Asserts.assertTrue(stackTrace.isTruncated());
-            int count = 0;
-            for (RecordedFrame frame : frames) {
-                Asserts.assertTrue(frame.isJavaFrame());
-                Asserts.assertNotNull(frame.getMethod());
-                RecordedMethod m = frame.getMethod();
-                Asserts.assertNotNull(m.getType());
-                if (m.getName().contains(stackMethod)) {
-                    count++;
+            Asserts.assertFalse(events.isEmpty(), "No event found in virtual thread");
+            for (RecordedEvent event : events) {
+                System.out.println(event);
+                RecordedStackTrace stackTrace = event.getStackTrace();
+                if (stackTrace == null) {
```

Comment:
This is interesting. Did you see events that did not have a stacktrace associated? jdk.ObjectAllocationOutsideTLAB?

Reply:
If that is the case, my hypothesis is that there are allocation sites that do not set the ljf, and therefore the stack trace code early out with a 0. Need to check with -XX:-UseTLAB, though.

Reply:
Yes, comes from native compiler allocations. Compiler threads are "Java threads" (you can even see them in `jstack`), but they don't have the Java stack. So you have the events like:

```
jdk.ObjectAllocationOutsideTLAB {
  startTime = 18:05:34.023 (2022-11-24)
  objectClass = java.lang.String (classLoader = bootstrap)
  allocationSize = 24 bytes
  eventThread = "C1 CompilerThread0" (javaThreadId = 25)
}
```